src/components/CompareResults/ResultsTable.tsx

```diff
@@ -107,20 +107,52 @@ export default function ResultsTable() {
     new Map() as Map<string, Set<string>>, // ColumnID -> Set<Values to remove>
   );
 
-  const onClearFilter = (columnId: string) => {
-    setTableFilters((oldFilters) => {
-      const newFilters = new Map(oldFilters);
-      newFilters.delete(columnId);
-      return newFilters;
+  useEffect(() => {
+    const initialFilters = new Map<string, Set<string>>();
+
+    cellsConfiguration.forEach(({ key, possibleValues }) => {
+      if (!possibleValues) return;
+
+      const paramsValue = rawSearchParams.get(key) ?? possibleValues.join(',');
+      initialFilters.set(
+        key,
+        new Set(possibleValues.filter((v) => !paramsValue.includes(v))),
+      );
+
+      if (!rawSearchParams.has(key)) rawSearchParams.set(key, paramsValue);
     });
+
+    updateRawSearchParams(rawSearchParams);
+    setTableFilters(initialFilters);
+  }, []);
+
+  const onClearFilter = (columnId: string) => {
+    const possibleValues = cellsConfiguration.find(
+      (c) => c.key === columnId,
+    )?.possibleValues;
+
+    if (!possibleValues) return;
+
+    setTableFilters((filters) => (filters.delete(columnId), new Map(filters)));
+    rawSearchParams.set(columnId, possibleValues.join(','));
+    updateRawSearchParams(rawSearchParams);
   };
 
   const onToggleFilter = (columnId: string, filters: Set<string>) => {
-    setTableFilters((oldFilters) => {
-      const newFilters = new Map(oldFilters);
-      newFilters.set(columnId, filters);
-      return newFilters;
-    });
+    const possibleValues = cellsConfiguration.find(
+      (c) => c.key === columnId,
+    )?.possibleValues;
+
+    if (!possibleValues) return;
+
+    const paramsValue =
+      filters.size === possibleValues.length
+        ? ''
+        : possibleValues.filter((v) => !filters.has(v)).join(',');
+
+    setTableFilters((prev) => new Map(prev).set(columnId, filters));
+    rawSearchParams.set(columnId, paramsValue);
+    updateRawSearchParams(rawSearchParams);
```

Comment:
This looks like a great solution. Good job on your first attempt and thanks for the detailed PR description! However, would you mind using shorthand or whole words instead of initials ? For example : `cell vs `c` etc? Otherwise, I think this would be a good PR to be merged.

Reply:
Okay, I will do that. Thank you!